code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,298 @@
+/obj/item/inflatable
+	name = "inflatable"
+	w_class = 2
+	icon = 'icons/obj/inflatable.dmi'
+	var/deploy_path = null
+
+/obj/item/inflatable/attack_self(mob/user)
+	if(!deploy_path)
+		return
+	playsound(loc, 'sound/items/zip.ogg', 75, 1)
+	user << "<span class='notice'>You inflate \the [src].</span>"
+	var/obj/structure/inflatable/R = new deploy_path(user.loc)
+	src.transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	qdel(src)
+
+/obj/item/inflatable/wall
+	name = "inflatable wall"
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation."
+	icon_state = "folded_wall"
+	deploy_path = /obj/structure/inflatable/wall
+
+/obj/item/inflatable/door/
+	name = "inflatable door"
+	desc = "A folded membrane which rapidly expands into a simple door on activation."
+	icon_state = "folded_door"
+	deploy_path = /obj/structure/inflatable/door
+
+/obj/structure/inflatable
+	name = "inflatable"
+	desc = "An inflated membrane. Do not puncture."
+	density = 1
+	anchored = 1
+	opacity = 0
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "wall"
+	penetration_dampening = 3
+
+	var/undeploy_path = null
+	var/health = 50.0
+
+/obj/structure/inflatable/wall
+	name = "inflatable wall"
+	undeploy_path = /obj/item/inflatable/wall
+
+/obj/structure/inflatable/New(location)
+	..()
+	update_nearby_tiles(need_rebuild=1)
+
+/obj/structure/inflatable/Destroy()
+	update_nearby_tiles()
+	..()
+
+/obj/structure/inflatable/Cross(atom/movable/mover, turf/target, height=0, air_group=0)
+	return 0
+
+/obj/structure/inflatable/bullet_act(var/obj/item/projectile/Proj)
+
+	health -= Proj.damage
+	..()
+	if(health <= 0)
+		deflate(1)
+	return
+
+/obj/structure/inflatable/projectile_check()
+	return PROJREACT_WINDOWS
+
+/obj/structure/inflatable/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			qdel(src)
+			return
+		if(2.0)
+			deflate(1)
+			return
+		if(3.0)
+			if(prob(50))
+				deflate(1)
+				return
+
+/obj/structure/inflatable/attack_hand(mob/user as mob)
+	user.delayNextAttack(10)
+	add_fingerprint(user)
+	return
+
+/obj/structure/inflatable/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	user.delayNextAttack(10)
+	if (W.is_sharp())
```

Comment:
extra space between if and the parenthesis, inconsistent with the rest of the code.